### ISSUE TITLE
figure out what to about the shapeless code in the parsing module

### ISSUE THREAD

**Issue by pjfanning** [apache/incubator-pekko-http#63]
* relates to #35 
* Unlike #1, there is no decision to remove the shapeless code yet.
* it could be that if #1 is merged, that we no longer need the shapeless code (I haven't checked if the parboiled2 code is the only code using the shapeless code)
* shapeless3 exists for Scala 3 but does include HList - I guess the logic is that Scala 3 tuples can do the task
* so if we do use HLIst or the other shapeless code in pekko-http (after #1 is merged or deferred) then we might want to keep the shapeless code
* if we keep it, we will need to consider updating our LICENSE and NOTICE. 
   * https://github.com/milessabin/shapeless has no NOTICE
   * it is Apache licensed so we may need to document it in our license (this needs to be verified) 
* relates to #62

**Comment by mdedetrich** [comment]
Regarding Scala 3, Scala has apparently added HLists and other datastructures which shapeless has historically been used for (see https://www.scala-lang.org/2021/02/26/tuples-bring-generic-programming-to-scala-3.html), so one thing we can do is to remove shapeless for Scala 3 specifically

**Comment by pjfanning** [comment]
this code was removed alongside the parboiled code
